### DISCUSSION TITLE
Jetson Xavier NX dev kit not booting.

### DISCUSSION THREAD

**Discussion by jkdtgit** [OE4T/meta-tegra#904] [General]
Hi,
I have built the yocto image for Jetson Xavier NX dev kit (using SD card)

**git clone -b honister https://github.com/OE4T/tegra-demo-distro.git
cd tegra-demo-distro/
git branch -a
git submodule update --init
. ./setup-env --machine jetson-xavier-nx-devkit
bitbake demo-image-full**

I flashed it using 

**sudo ./doflash.sh**

The flashing was successful.
But after flashing the board does not boot.

I also tried by preparing the SD card image using 
**$ ./dosdcard.sh <filename>**

Then write the SD card with the above SD card image, using dd command. But it did not boot.

ref page: https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit

I have not customized anything. Just built and trying the same to boot.

What should be checked on this?

Thanks,
jk

**Comment by madisongh** [comment]
You should get better results if you have the `dosdcard.sh` script write directly to your SDcard (just specify the device name of the card reader/writer on the command line). If you really need the intermediate SDcard image as a file, make sure that you're creating an image that is the right size for the SDcard you're trying to write to.  It also doesn't hurt to run `fsck` on the rootfs partitions once you've copied the image to it.

**Comment by jkdtgit** [comment]
I have used ./dosdcard.sh <file name> to create the SD card image file.
Then I copied that image file to SD card with "sudo dd..." command. Then the SD card is put back on the board, but it does not boot.
I dont see any console messages.

**Reply by madisongh** [reply]
Can you at least try having dosdcard.sh write directly to the SDcard instead of a file, as I described above, and see if it works better?

You *should* see console messages (on the serial console) regardless of whether or not you have an SDcard in place, since the Xavier NX bootloaders are the SPI flash.  Do you not have a serial console connected?

**Reply by jkdtgit** [reply]
HI Madisongh,
I tried with sudo ./dosdcard.sh /dev/mmcblk0, and I get
"Init...ERR: could not initialize GPT on /dev/mmcblk0" error.

**Reply by madisongh** [reply]
A couple of things to check:

* Is `/dev/mmcblk0` the name of the SDcard writer device on your development host?
* Does the OS on your development system automatically mount SDcards or other removable devices on insertion?  That will interfere with the script's ability to format the card (and could also explain the problem you were having with `dd`ing the image)... Make sure the device is not in use by anything else on the system, either by disabling auto-mounts of removable devices, or unmounting the partitions before trying to use the script.

**Reply by jkdtgit** [reply]
Hi Madisongh,
I unmounted SD card and flashed again. it worked. Thanks.
